### PR TITLE
Issue # 152 borsuk performance

### PR DESCRIPTION
**Pull request by mjaniakcds**

### CODE REVIEW

File: src/main/scala/carldata/borsuk/Main.scala [hunk 1]
```diff
@@ -35,6 +35,14 @@ object Main {
 
     // HTTP listener will run in main thread
     Log.info("Server started on port 8080.")
+    // memory info
+    val mb = 1024*1024
```
Comment (wenta) [inline src/main/scala/carldata/borsuk/Main.scala]: I am not sure if this (and next lines) should be here.

File: src/main/scala/carldata/borsuk/envelope/Envelope.scala [hunk 2]
```diff
@@ -52,6 +54,7 @@ class Envelope(modelType: String, id: String) {
   }
 
   def list(): HashMap[String, EnvelopeResult] = {
+    Log.info("List for model: " + this.id)
```
Comment (wenta) [inline src/main/scala/carldata/borsuk/envelope/Envelope.scala]: Maybe not `info`, but `debug`? (here and in another places) If we leave it this way, after one week of "normal" work we will have a lot of unnecessary logs.

File: src/main/scala/carldata/borsuk/helper/SizeEstimator.scala [hunk 3]
```diff
@@ -0,0 +1,260 @@
+package carldata.borsuk.helper
```
Comment (wenta) [inline src/main/scala/carldata/borsuk/helper/SizeEstimator.scala]: This object comes from external library? Is there any way to add it as dependency and not add only single file?

File: src/main/scala/carldata/borsuk/rdiis/RDII.scala [hunk 4]
```diff
@@ -26,35 +28,49 @@ class RDII(modelType: String, id: String) {
...
       val rainfall: TimeSeries[Double] = TimeSeries(indexRainfall.toVector, params.rainfall.values.toVector).filter(_._2 >= 0)
       val minSessionWindow = if (params.minSessionWindow == Duration.ZERO) rainfall.resolution else params.minSessionWindow
 
-      val ts = rainfall.slice(rainfall.index.head, dtToInstant(edRainfall)).join(flow.slice(rainfall.index.head, dtToInstant(edRainfall)))
+      val ts = TimeSeriesHelper.slice(rainfall, rainfall.index.head, dtToInstant(edRainfall))
+        .join(TimeSeriesHelper.slice(flow, rainfall.index.head, dtToInstant(edRainfall)))
 
       val rainfall2 = TimeSeries(ts.index, ts.values.map(_._1))
 
+      // memory info
```
Comment (wenta) [inline src/main/scala/carldata/borsuk/rdiis/RDII.scala]: If this code is unnecessary please remove it.

File: src/main/scala/carldata/borsuk/rdiis/RDII.scala [hunk 5]
```diff
@@ -151,46 +173,55 @@ case class RDIIBuilder(rainfall: TimeSeries[Double], flow: TimeSeries[Double], s
...
+          TimeSeriesHelper.slice(rainfall,
+            startDate.minusMonths(3).toInstant(ZoneOffset.UTC), shiftedEd.plus(1, ChronoUnit.HOURS))
+            .groupByTime(_.truncatedTo(ChronoUnit.HOURS), _.map(_._2).sum)
+            .addMissing(dwp.resolution, (_, _, _) => 0.0)
+            .filter(x => dwp.index.contains(x._1))
+          , shiftedSd, shiftedEd)
+
+      val flowSection: TimeSeries[Double] = TimeSeriesHelper.slice(flow, shiftedSd, shiftedEd).filter(x => dwp.index.contains(x._1))
+
+      //Log.info("flowSection: " + SizeEstimator.estimate(flowSection))
```
Comment (wenta) [inline src/main/scala/carldata/borsuk/rdiis/RDII.scala]: If we do not need this comment please remove it.

File: src/test/scala/borsuk/EnvelopeApiTest.scala [hunk 6]
```diff
@@ -242,7 +242,7 @@ class EnvelopeApiTest extends WordSpec with Matchers with ScalatestRouteTest wit
         )
         fitEnvelopeRequest("test-id", fitEnvelopeParams) ~> route ~> check {
           status shouldBe StatusCodes.OK
-          eventually(timeout(120.seconds), interval(2.seconds)) {
+          eventually(timeout(220.seconds), interval(2.seconds)) {
```
Comment (wenta) [inline src/test/scala/borsuk/EnvelopeApiTest.scala]: Does it necessary?

File: src/main/scala/carldata/borsuk/Main.scala (hunk 1 above)
Comment (mjaniakcds) [reply src/main/scala/carldata/borsuk/Main.scala]: Deleted

File: src/main/scala/carldata/borsuk/envelope/Envelope.scala (hunk 2 above)
Comment (mjaniakcds) [reply src/main/scala/carldata/borsuk/envelope/Envelope.scala]: Changed to debug

File: src/main/scala/carldata/borsuk/helper/SizeEstimator.scala (hunk 3 above)
Comment (mjaniakcds) [reply src/main/scala/carldata/borsuk/helper/SizeEstimator.scala]: Yes, but the external library is no compatible with scala 2.12

File: src/test/scala/borsuk/EnvelopeApiTest.scala (hunk 6 above)
Comment (mjaniakcds) [reply src/test/scala/borsuk/EnvelopeApiTest.scala]: Changed

**Pull request merged**